https://issues.redhat.com/browse/ACM-16502 (#7567)
open-cluster-management/rhacm-docs: global_hub/global_hub_upgrade_postgres.adoc

[#global-hub-upgrade-postgres]
= Recovering data from the upgraded built-in Postgres

Starting with {global-hub} version 1.4.0, the built-in Postgres database gets upgraded to version 16. This upgrade replaces the earlier `multicluster-global-hub-postgres` instance with the new `multicluster-global-hub-postgressql` instance. By default, this upgrade automatically re-syncs real-time data, such as: `policies` and `clusters`, to the new Postgres instance. However, historical data, such as: `event` and `history` tables, does not automatically get recovered. 

If you want to keep historical data, complete the following sections: 

* <<restoring-data, Restoring historical data>>
* <<deleting-legacy-postgres, Deleting legacy built-in postgres data>>

[#restoring-data]
== Restoring historical data

When your historical data gets removed by the {global-hub} upgrade to version 1.4.0, recover your data by completing the following steps:

. Restore your history tables by running the following script: 

+
[source,bash]
----

> oc ./doc/upgrade/restore_history_tables.sh

Reviewer: The shell script provided  https://github.com/stolostron/multicluster-global-hub/tree/release-2.13/doc/upgrade, you need to provide the shell here or use the following script to clone the globalhub repo and run the shell.

```
git clone https://github.com/stolostron/multicluster-global-hub.git
sh ./doc/upgrade/restore_history_tables.sh
```

All the following script should do in the same way.

[#global-hub-upgrade-postgres]
= Recovering data from the upgraded built-in Postgres

> Starting with {global-hub} version 1.4.0, the built-in Postgres database gets upgraded to version 16. This upgrade replaces the earlier `multicluster-global-hub-postgres` instance with the new `multicluster-global-hub-postgressql` instance. By default, this upgrade automatically re-syncs real-time data, such as: `policies` and `clusters`, to the new Postgres instance. However, historical data, such as: `event` and `history` tables, does not automatically get recovered. 

Reviewer 2: ```suggestion
Starting with {global-hub} version 1.4.0, the built-in Postgres database is upgraded to version 16. This upgrade replaces the earlier `multicluster-global-hub-postgres` instance with the new `multicluster-global-hub-postgressql` instance.

By default, this upgrade automatically re-syncs real-time data, such as `policies` and `clusters`, to the new Postgres instance. Historical data, such as `event` and `history` tables, are not automatically recovered.
```

If you want to keep historical data, complete the following sections: 

* <<restoring-data, Restoring historical data>>
* <<deleting-legacy-postgres, Deleting legacy built-in postgres data>>

[#restoring-data]
== Restoring historical data

> When your historical data gets removed by the {global-hub} upgrade to version 1.4.0, recover your data by completing the following steps:

Reviewer 2: ```suggestion
When the {global-hub} upgrade to version 1.4.0 removes your historical data, recover your data by completing the following steps:
```

. Close the {acm} Git repo to access the shell script that you need to recover your data by running the following command: 

+
[source,bash]
----
oc git clone https://github.com/stolostron/multicluster-global-hub.git
----

. Restore your history tables by running the following shell script: 

+
[source,bash]
----

> oc ./doc/upgrade/restore_history_tables.sh

Reviewer: It should be 
`sh ./doc/upgrade/restore_history_tables.sh`
or 
`./doc/upgrade/restore_history_tables.sh`

+
[source,bash]
----
./doc/upgrade/restore_history_tables.sh
----

.. If the namespace is not specified as `multicluster-global-hub`, set the {global-hub} namespace when you run the following shell script: 

+
[source,bash]
----
./doc/upgrade/restore_history_tables.sh <1>
----
<1> The {global-hub} that you installed. 

. Restore your event tables by completing the following steps: 

.. If the default namespace is `multicluster-global-hub`, run the following shell script: 

+
[source,bash]
----
./doc/upgrade/restore_event_tables.sh
----

.. If the namespace is not specified as `multicluster-global-hub`, set the {global-hub} namespace when you run the following shell script: 

+
[source,bash]
----
./doc/upgrade/restore_event_tables.sh <1>
----
<1> The {global-hub} that you installed. 

[#deleting-legacy-postgres]
== Deleting legacy built-in Postgres data

After {global-hub} upgrades to version 1.4.0, it switches to the new built-in Postgres instance. The global hub operator does not automatically delete resources associated with the legacy Postgres instance. To delete the legacy Postgres resources, run the following script: 

[source,bash]
----

> ./doc/upgrade/cleanup_legacy_postgres.sh

Reviewer 3: Hi @jc-berger, this command is same with above commands, need to set the globalhub namespace when running it.

+
[source,bash]
----
git clone https://github.com/stolostron/multicluster-global-hub.git
----

. Restore your history tables by running the following shell script: 

+
[source,bash]
----

> ./doc/upgrade/restore_history_tables.sh

Reviewer 3: Need to set the globalhub installed namespace when run the script. if it's the default namespace `multicluster-global-hub`, can run the script directly, otherwise, run as below
```
./doc/upgrade/restore_history_tables.sh <your_installed_globalhub_namespace>
```
The same method for all shell scripts.
